mllib/src/main/scala/org/apache/spark/ml/optim/WeightedLeastSquares.scala

```diff
@@ -108,6 +101,53 @@ private[ml] class WeightedLeastSquares(
           "Consider setting fitIntercept=true.")
       }
     }
+    /*
+      If more than of the features in the data are constant (i.e, data matrix has constant columns),
+      then A^T.A is no longer positive definite and Cholesky decomposition fails (because the
+      normal equation does not have a solution).
+      In order to find a solution, we need to drop constant columns from the data matrix. Or,
+      we can drop corresponding column and row from A^T.A matrix.
+      Once we drop rows/columns from A^T.A matrix, the Cholesky decomposition will produce
+      correct coefficients. But, for the final result, we need to add zeros to the list of
+      coefficients corresponding to the constant features.
+   */
+    val aVarRaw = summary.aVar.values
+    // this will keep track of features to keep in the model, and remove
+    // features with zero variance.
+    val nzVarIndex = aVarRaw.zipWithIndex.filter( _._1 != 0 ).map( _._2 )
```

Comment:
I don't think you need the leading and trailing spaces inside the filter and map.